sklearn/utils/tests/test_extmath.py

```diff
@@ -452,6 +454,99 @@ def naive_log_logistic(x):
     assert_array_almost_equal(log_logistic(extreme_x), [-100, 0])
 
 
+@pytest.fixture()
+def rng():
+    return np.random.RandomState(42)
```

Comment:
I don't think it's a good idea. RandomState is mutable, and so the random matrices would change depending on the order in which these tests are run. It would be better to re-initialize it in each function that needs it.

Reply:
@rth indeed I took that into account. I agree with you that tests must be reproducible and thus seeded random number generator must be re-initialized at each function. But that is exactly the way I implemented it. The default fixture scope is `'function'`, and thus it is invoked once per test function.

You may wonder why I have factorized out the initialization of the random number generator if there is no performance gain. And the answer is that we eventually should evolve from slow legacy Mersenne Twister pseudo-random number generator (`np.random.RandomState`) to the new PCG-64 pseudo-random number generator (`np.random.default_rng`).  And this would be much simpler by modifying just the fixture code.

Reply:
Yes, you are right. Nevermind my comment.